packages/py-moose-lib/moose_lib/dmv2/materialized_view.py

```diff
@@ -5,36 +5,44 @@
 including their SQL statements, target tables, and dependencies.
 """
 from typing import Any, Optional, Union, Generic, TypeVar
-from pydantic import BaseModel, ConfigDict
+from pydantic import BaseModel, ConfigDict, model_validator
 
 from moose_lib import ClickHouseEngines
 from .types import BaseTypedResource, T
 from .olap_table import OlapTable, OlapConfig
 from .sql_resource import SqlResource
 
+
 class MaterializedViewOptions(BaseModel):
     """Configuration options for creating a Materialized View.
 
     Attributes:
         select_statement: The SQL SELECT statement defining the view's data.
         select_tables: List of source tables/views the select statement reads from.
-        table_name: The name of the underlying target table storing the materialized data.
+        table_name: (Deprecated in favor of target_table) Optional name of the underlying
+                    target table storing the materialized data.
         materialized_view_name: The name of the MATERIALIZED VIEW object itself.
-        engine: Optional ClickHouse engine for the target table.
+        engine: Optional ClickHouse engine for the target table (used when creating
+                a target table via table_name or inline config).
         order_by_fields: Optional ordering key for the target table (required for
                          engines like ReplacingMergeTree).
+        target_table: Either an existing `OlapTable` instance to use as the destination,
```

Comment:
This is actually not an option on this no?

Reply:
fixed